### ISSUE TITLE
Update to Metadata 1.0 format

### ISSUE THREAD

**Issue by abadger** [ansible/proposals#54]
# Proposal: Minor update to metadata 1.0
*Author*: Toshio Kuratomi<@abadger> IRC: abadger1999

*Date*: 2016/02/27

- Status: New
- Proposal type: module management
- Targeted Release: 2.3
- Associated PR: <link to GH PR in ansible/proposals if PR was submitted>
- Estimated time to implement: 1 week

## Motivation
Due to a misunderstanding when finalizing the module metadata in https://github.com/ansible/proposals/issues/30 there is a value for the supported_by field called committer.  All of our documentation uses the more descriptive name "curated" for it.   Before 2.3 goes out and we're stuck with the disagreement between publicly visible docs and the internal metadata, let's fix that.

There's also a few other cleanups of metadata that could be done at the same time: version confuses module authors and the maintained/unmaintained information is better done outside of a module (as this can change much more rapidly than we make releases).

### Problems
What problems exist that this proposal will solve?
* The documentation talks about modules which are curated by the ansible team.  But a module contributor will see "committer" inside of the metadata instead of "curated".
* The "version" field keeps being mistaken for the version of the module or of ansible instead of the version of the metadata. 
* Whether a module has an active owner changes more rapidly than we release modules.  So the unmaintained field in modules is not to be relied upon.

## Solution proposal
* Rename the committer entry of supported_by to curated.
* Remove the unmaintained entry of supported_by
* Rename the version field to metadata_version.

The new metadata will look like these examples:
```
# Data formatted as a python dict:
ANSIBLE_METADATA = {
    'metadata_version': '1.0',
    'supported_by': 'core',
    'status': ['stableinterface', 'deprecated']
}

# Data formatted as a yaml string:
ANSIBLE_METADATA = r"""
    metadata_version: '1.0'
    supported_by: 'core'
    status: ['stableinterface', 'deprecated']
"""
```

In documentation the following descriptions will change:
The version field will now be the metadata_version field:
* **metadata_version**: An "X.Y" formatted string. X and Y are integers which define the metadata format version. Modules shipped with Ansible are tied to an Ansible release so we will only ship with a single version of the metadata. We'll increment Y if we add fields or legal values to an existing field. We'll increment X if we remove fields or values or change the type or meaning of a field.

supported_by will have the unmaintained entry removed and the committer will be reworded to be curated like this:
* **curated**: The ansible core team is the gatekeeper for this module. They will review PRs from the community before merging but might not generate fixes and code for new features on their own.

The default metadata should now be :
```
ANSIBLE_METADATA = {'metadata_version': '1.0',
                                        'supported_by': 'community',
                                        'status': ['preview']}
```

## Dependencies (optional)
* The hacking/metadata-tool.py script needs to be updated
* All ansible modules will need to be refreshed with the new metadata format
* The module formatter for generating the website will need to be checked  It may have some special logic for mapping committer to curated that needs to be removed.  May need updated wording as well.
* ansible-doc may need <strike>to be updated to remove any special logic for mapping committer to curated and may need</strike> updated wording
* there is a test that ANSIBLE_METADATA contains valid information in validate-modules.   The schema for ANSIBLE_METADATA will need to be updated which is a trivial change.

## Documentation (optional)
* The dev_guide/developing_modules_documenting.rst will need to be updated to document the new field, values, and default.

## Alternatives
* We debated whether to up the version (since the metadata has been in for several months now) but decided not to because it hasn't gone into an ansible release yet.

**Comment by bcoca** [comment]
ansible-doc should not need any updates as it only displays actual metadata fields, it has no mappings

**Comment by sivel** [comment]
`validate-modules` holds a schema for `ANSIBLE_METADATA` and will need to be updated which is a trivial change.

**Comment by bcoca** [comment]
in general users should be able to add their own keys, so runtime tools should not validate it too deeply, validate-modules is for inclusion in the repo so it does make sense it does.

**Comment by abadger** [comment]
Updated.  For the testing dependency, added the specific info that validate-modules is what needs changing.

**Comment by abadger** [comment]
This has now been merged into devel for 2.3: https://github.com/ansible/ansible/issues/22587
